parsec/serde/schema.py

```diff
@@ -201,3 +208,19 @@ def validate(self, data, many=None, partial=None):
             return self.load(data, many=many, partial=partial).errors
         except ValidationError as ve:
             return ve.messages
+
+
+class OneOfSchemaLegacy(OneOfSchema):
+    _instantiated_schemas = None
+    _instantiated_fallback_schema = None
+
+    def _get_schema(self, type):
+        if self._instantiated_schemas is None:
+            self._instantiated_schemas = {
+                k: v if isinstance(v, Schema) else v() for k, v in self.type_schemas.items()
+            }
+
+            if self.fallback_type_schema and not isinstance(self.fallback_type_schema, Schema):
```

Comment:
If we have `self.fallback_type_schema = Myschema()`, then we don't go in this if, hence `self._instantiated_fallback_schema` is`None`...

Reply:
I removed `and not isinstance(self.fallback_type_schema, Schema)`, do you think we shall have some extra assertion on the type ?

Reply:
I think it's fine to consider `self.fallback_type_schema` must be a `Schema`, and in such case we don't use assert to enforce this, considering it's a static typing job to make this check

besides, if we pass an instance of schema instead of a schema class, we will have an exception anyway when trying to `instantiate` the instance (well except if the instance is callable, but then the user has really really fucked up ^^)